memcached.c

```diff
@@ -3440,19 +3467,34 @@ static inline void process_get_command(conn *c, token_t *tokens, size_t ntokens,
                 }
 
                 /* item_get() has incremented it->refcount for us */
-                pthread_mutex_lock(&c->thread->stats.mutex);
-                c->thread->stats.lru_hits[it->slabs_clsid]++;
-                c->thread->stats.get_cmds++;
-                pthread_mutex_unlock(&c->thread->stats.mutex);
+                if (should_touch) {
+                    pthread_mutex_lock(&c->thread->stats.mutex);
```

Comment:
There isn't really contention on these locks, so it's okay to put them outside of the branches and save a few lines.